src/WebJobs.Extensions.OpenAI/Agents/ChatBotService.cs

```diff
@@ -98,39 +98,51 @@ public async Task CreateChatBotAsync(ChatBotCreateRequest request, CancellationT
         await this.tableClient.CreateIfNotExistsAsync();
 
         // Check to see if the chat bot has already been initialized
-        Pageable<TableEntity> queryResultsFilter = this.tableClient.Query<TableEntity>(filter: $"PartitionKey eq '{request.Id}'");
+        AsyncPageable<TableEntity> queryResultsFilter = this.tableClient.QueryAsync<TableEntity>(filter: $"PartitionKey eq '{request.Id}'");
 
-        if (queryResultsFilter.Any())
-        {
-            // Create a batch of table transaction actions for deleting entities
-            List<TableTransactionAction> deleteBatch = new List<TableTransactionAction>();
+        // Create a batch of table transaction actions for deleting entities
+        List<TableTransactionAction> deleteBatch = new();
+
+        // Count of entities to delete to keep track of
+        int count = 0;
 
-            // Get all entities with the same partition key and add to batch for deletion
-            foreach (TableEntity entity in queryResultsFilter)
+        await foreach (TableEntity entity in queryResultsFilter)
+        {
+            // If the count is greater than or equal to 100, submit the transaction and clear the batch
+            if (count >= 100)
```

Comment:
Do we really need a `count` variable or could we simplify and just use `deleteBatch.Count`?

Reply:
nope removed count!